src/containers/App.js

```diff
@@ -0,0 +1,98 @@
+// src/containers/App.js
+
+import React, { Component } from 'react'
+
+import { ALL_TODOS, ACTIVE_TODOS, COMPLETED_TODOS } from '../constants';
+import Footer from '../components/Footer';
+import Header from '../components/Header';
+import Todos from '../components/Todos';
+import Todo from '../components/Todo';
+
+class App extends Component {
+	constructor(props) {
+		super(props);
+
+		// JSX supports passing in element arrays
+		/*
+		const array = [
+			<Todo label="First todo..." deleteHandler={this.handleTodoDelete} />,
+			<Todo label="Second todo..." deleteHandler={this.handleTodoDelete} />,
+			<Todo label="Third todo..." deleteHandler={this.handleTodoDelete} />
+		];
+		*/
+		this.filter = this.filter.bind(this);
+		this.handleTodoAdd = this.handleTodoAdd.bind(this);
+		this.handleTodoClear = this.handleTodoClear.bind(this);
+		this.handleTodoDelete = this.handleTodoDelete.bind(this);
+		this.handleTodoComplete = this.handleTodoComplete.bind(this);
+
+		this.state = {
+			lastIndex: 1,
+			show: ALL_TODOS,
+			items: []
+		};		
+	}
+
+	handleTodoAdd(e) {
+		console.log('add item');
+
+		if (e.target.value == '') return 0;
+
+    		if (e.key === 'Enter') {
+			this.setState({ lastIndex : this.state.lastIndex + 1 });
+      			this.setState({ items : this.state.items.concat(<Todo key={this.state.lastIndex} completed={false} label={e.target.value} deleteHandler={this.handleTodoDelete} completeHandler={this.handleTodoComplete}/>)});
```

Comment:
In JavaScript concat is a method that is non-destructive, meaning it will return a brand new array rather than mutating the original.  

React + Redux focus on functional programming which immutability is a big part of.  concat is definitely the right method to use here (as opposed to push).  You may also see other developers use the ES6 spread operator to concat arrays:
```
const newArray = [...oldArray, newItem];
```